### PR TITLE
Adding default boolean check

### PR DESCRIPTION
**Pull request by bmalinconico**
Adds check for default boolean config

### CODE REVIEW

File: spec/dynamoid_advanced_where/node_tests/equality_spec.rb [hunk 1]
```diff
@@ -32,11 +33,25 @@
...
     end
 
-    it "matches true" do
+    it "matches false" do
       expect(klass.where{ native_bool == false }.all).to match_array([item2])
     end
   end
 
+  describe "default boolean equality" do
+    let!(:item1) { klass.create(default_bool: true) }
```
Comment (nrowegt) [inline spec/dynamoid_advanced_where/node_tests/equality_spec.rb]: `true_item` `false_item` `nil_item` makes specs clearer and easier to read, not a blocker for merge.

File: spec/dynamoid_advanced_where/node_tests/equality_spec.rb [hunk 2]
```diff
@@ -32,11 +33,25 @@
...
+    let!(:item2) { klass.create(default_bool: false) }
+    let!(:item3) { klass.create(default_bool: nil) }
+
+    it "matches true" do
+      expect(klass.where{ default_bool == true }.all).to match_array([item1])
+    end
```
Comment (nrowegt) [inline spec/dynamoid_advanced_where/node_tests/equality_spec.rb]: should it also have a case for matches NIL/DNE?

File: spec/dynamoid_advanced_where/node_tests/equality_spec.rb (hunk 1 above)
Comment (bmalinconico) [reply spec/dynamoid_advanced_where/node_tests/equality_spec.rb]: agreed I'll update them

File: spec/dynamoid_advanced_where/node_tests/equality_spec.rb (hunk 2 above)
Comment (bmalinconico) [reply spec/dynamoid_advanced_where/node_tests/equality_spec.rb]: There is a nil item which doesn't match true or false. Can you clarify the other case you think would be helpful?

Comment (nrowegt) [reply spec/dynamoid_advanced_where/node_tests/equality_spec.rb]: I expected that testing equality for `nil` would also be relevant, but doesn't look like that's the case.

Comment (bmalinconico) [reply spec/dynamoid_advanced_where/node_tests/equality_spec.rb:48]: the test for nil is the existence spec spec/dynamoid_advanced_where/node_tests/exists_spec.rb

The tests are organized by operator not by field type, which may or may not be correct :/

**Pull request merged**